modules/ecs-service/main.tf

```diff
@@ -0,0 +1,692 @@
+# LOOKUP IMAGE FOR ECS SERVICE
+# -------------------------------------------
+
+# This will get the ECS service that is currently deployed.
+data "aws_ecs_service" "service" {
+  count = !var.create_scheduled_task && local.lookup_deployed_image ? 1 : 0
+
+  cluster_arn  = data.aws_ecs_cluster.cluster.arn
+  service_name = var.ecs_service_name
+}
+
+# This will get the latest container definition from the task definition
+# that is currently deployed to the ECS service.
+data "aws_ecs_container_definition" "service" {
+  count = !var.create_scheduled_task && local.lookup_deployed_image ? 1 : 0
+
+  task_definition = data.aws_ecs_service.service[0].task_definition
+  container_name  = var.ecs_service_name
+}
+
+
+# -------------------------------------------
+# LOOKUP IMAGE FOR SCHEDULED TASK
+# -------------------------------------------
+
+# Gets the latest container definition from the max
+# revision of the task definition.
+data "aws_ecs_container_definition" "scheduled" {
+  count = var.create_scheduled_task && local.lookup_deployed_image ? 1 : 0
+
+  task_definition = local.task_definition
+  container_name  = var.ecs_service_name
+
+  depends_on = [
+    data.aws_ecs_task_definition.task
+  ]
+}
+
+
+# -------------------------------------------
+# CREATE THE ECS TASK DEFINITION
+# -------------------------------------------
+
+locals {
+  cloudwatch_log_group_name = "/ecs/${var.ecs_service_name}"
+  log_configuration = {
+    logDriver = "awslogs"
+    options = {
+      awslogs-group  = local.cloudwatch_log_group_name
+      awslogs-region = data.aws_region.current.name
+    }
+  }
+
+  repositoryCredentials = {
+    credentialsParameter = var.docker_credential_secretsmanager_arn
+  }
+}
+
+resource "aws_ecs_task_definition" "task" {
+  family = var.ecs_service_name
+
+  cpu          = 256
+  memory       = 256
+  network_mode = var.create_scheduled_task ? "awsvpc" : "bridge"
+
+  execution_role_arn = aws_iam_role.task_execution.arn
+  task_role_arn      = length(var.ecs_task_role_policy_arns) > 0 ? aws_iam_role.task.arn : null
+
+  container_definitions = jsonencode([
+    {
+      name = var.ecs_service_name
+
+      image = local.image
+
+      repositoryCredentials = var.docker_credential_secretsmanager_arn != "" ? local.repositoryCredentials : null
+
+      portMappings = [{
+        name          = random_id.service_connect[0].hex
+        containerPort = var.ecs_container_port
+      }]
+
+      # Environment Variables and Secrets are both string maps with
+      # the same key/value structure. They are mapped to the appropriate
+      # structure for the container definition
+      environment = [for k, v in var.ecs_container_environment_variables : { name = k, value = v }]
+      secrets     = [for k, v in var.ecs_container_secrets : { name = k, valueFrom = v }]
+
+      logConfiguration = var.enable_container_logs ? local.log_configuration : null
+
+      # healthCheck = !var.create_scheduled_task ? local.healthCheck : null
```

Comment:
Do we want to remove this?

Reply:
Unsure yet. What're your thoughts? Without it the only health check we have is ALB. For External services that is good but for internal services we could have services failing their checks but not being drained and replaced.